prov/gni/src/gnix_vc.c

```diff
@@ -1941,9 +1942,9 @@ static int __gnix_vc_push_work_reqs(struct gnix_vc *vc)
 			if ((ret != -FI_ENOSPC) &&
 			    (ret != -FI_EAGAIN)) {
 				/* TODO report error? */
-				GNIX_WARN(FI_LOG_EP_DATA,
-					  "Failed to push request %p: %s\n",
-					  req, fi_strerror(-ret));
+				GNIX_FATAL(FI_LOG_EP_DATA,
```

Comment:
> Why should this be fatal?

I actually don't think it should be now.  When testing these changes, an error on my part caused an incorrect memory handle to be set in the rdm transmission descriptor, triggering this conditional repeatedly until the program crashed.

It looks like we maintain (for rendezvous messages) a tx_failures count within the fabric request and drop the request if those failures exceed the max_retransmits domain parameter.

It looks like the UGNI library handles retransmission of Smsgs for certain systems but GNI_SmsgSendWTag is a non-blocking call so this might result in unexpected behavior in the rendezvous path later if the smsg never arrives or doesn't completely arrive, is there anything we can do to check for this kind or error, @ztiffany ?